Meta.Tests.ps1

```diff
@@ -12,23 +12,39 @@ $errorActionPreference = 'Stop'
 $testHelperModulePath = Join-Path -Path $PSScriptRoot -ChildPath 'TestHelper.psm1'
 Import-Module -Name $testHelperModulePath
 
-$moduleRootFilePath = Split-Path -Path $PSScriptRoot -Parent
-$moduleName = (Get-Item -Path $moduleRootFilePath).Name
-$dscResourcesFolderFilePath = Join-Path -Path $moduleRootFilePath -ChildPath 'DscResources'
-
 <#
-    This is a workaround to be able to test DscResource.Tests module.
-    Because the name has punctuation in the name, AppVeyor replaces
-    that with a dash when it creates the folder structure, so the folder
-    name becomes 'dscresource-tests'. This sets the module name to the
-    correct name.
-    If the name can be detected in a better way, for DscResource.Tests
-    and all other modules, then this could be removed.
+    This is a workaround to be able to test DscResource.Tests module. When
+    DscResource.Tests is testing itself, we need to determine if we are running
+    the code on repository DscResource.Tests or some other resource module.
+    if the $env:APPVEYOR_BUILD_FOLDER match the regex 'dscresource.tests' the
+    tests is considered that it should run on it self.
+    The regex will match both 'DscResource.Tests', when run locally, and
+    'dscresource-tests' when run in AppVeyor.
+
+    NOTE: To be able to run these common tests locally the environment variable
```

Comment:
Thanks for questioning this. 😄 It should have said "To be able to run these common tests locally **for the DscResource.Test module**...".
If we set the variable to 'DscResource.Test' then other resource modules than DscResource.Tests won't be able to run the common tests. 
But I agree this is bad. I didn't like the idea either when I sent it in, but thought that there is so few contributing to this repo so maybe we can live with this. But we shouldn't need to. 😄 

After seeing your comment, and during the walk to work this morning I thought more of an other idea I had, and that is to check if the parent folder contains a module manifest. The downside to this is that if the user puts any module manifest in the parent folder, than it will break testing of DscResource.Tests. But maybe _that_ is something we can live with 😉 

I will update with this logic, an try to be precise with the comments in the code as well. 😄